[17.0] mig-excel_import_export

Josep-s73 opened the pull request:
Standard migration 17.0

kittiu committed:
Excel Import/Export/Report  (#1522)

* [ADD] v12 excel_import_export

* Change from eval() to safe_evel()

* Change variable to format to style, as fomat is a common python function

:100644 100644 00ee3d9f... e9e48d87... M	excel_import_export/models/common.py
:100644 100644 a215d29b... 5b4d1fb1... M	excel_import_export/models/styles.py
:100644 100644 ace11a32... 01e5b9f5... M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f... cadfb0f2... M	excel_import_export/models/xlsx_import.py
:100644 100644 58689ee5... 80490ce8... M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6... a363ad19... M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 475b5187... 392fe6e5... M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 4af9c519... 45ee33c6... M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3... 17d3964d... M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f... 51c2572a... M	excel_import_export/models/common.py
:100644 100644 a215d29b... 5b4d1fb1... M	excel_import_export/models/styles.py
:100644 100644 ace11a32... 185a3330... M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f... cadfb0f2... M	excel_import_export/models/xlsx_import.py
:100644 100644 58689ee5... 80490ce8... M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6... a363ad19... M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 475b5187... 392fe6e5... M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 4af9c519... 45ee33c6... M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3... 933ce0dc... M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f... 51c2572a... M	excel_import_export/models/common.py
:100644 100644 a215d29b... 5b4d1fb1... M	excel_import_export/models/styles.py
:100644 100644 ace11a32... 185a3330... M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f... cadfb0f2... M	excel_import_export/models/xlsx_import.py
:100644 100644 58689ee5... 80490ce8... M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6... a363ad19... M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 475b5187... 392fe6e5... M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 4af9c519... 45ee33c6... M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3 3b1217e8 M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 5b4d1fb1 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 185a3330 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f cadfb0f2 M	excel_import_export/models/xlsx_import.py
:100644 100644 58689ee5 80490ce8 M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 475b5187 392fe6e5 M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3 fee958bc M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 5b4d1fb1 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 185a3330 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f cadfb0f2 M	excel_import_export/models/xlsx_import.py
:100644 100644 58689ee5 80490ce8 M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 475b5187 392fe6e5 M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3 fee958bc M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 9738a3c8 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 a7d6adc5 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f 12f9ca99 M	excel_import_export/models/xlsx_import.py
:100644 100644 70c37799 f123d2a6 M	excel_import_export/models/xlsx_report.py
:100644 100644 58689ee5 578a1fd8 M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 800ea573 1807ea7e M	excel_import_export/wizard/export_xlsx_wizard.py
:100644 100644 febed8d0 750dc17e M	excel_import_export/wizard/import_xlsx_wizard.py
:100644 100644 475b5187 392fe6e5 M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 8e40a2d0 21574896 M	excel_import_export_demo/report_sale_order/report_sale_order.py
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3 fee958bc M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 9738a3c8 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 c7db3f92 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f 12f9ca99 M	excel_import_export/models/xlsx_import.py
:100644 100644 70c37799 f123d2a6 M	excel_import_export/models/xlsx_report.py
:100644 100644 58689ee5 578a1fd8 M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 800ea573 1807ea7e M	excel_import_export/wizard/export_xlsx_wizard.py
:100644 100644 febed8d0 750dc17e M	excel_import_export/wizard/import_xlsx_wizard.py
:100644 100644 475b5187 392fe6e5 M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 8e40a2d0 21574896 M	excel_import_export_demo/report_sale_order/report_sale_order.py
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3 fee958bc M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 9738a3c8 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 c7db3f92 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f 12f9ca99 M	excel_import_export/models/xlsx_import.py
:100644 100644 70c37799 f123d2a6 M	excel_import_export/models/xlsx_report.py
:100644 100644 58689ee5 e3826e08 M	excel_import_export/models/xlsx_template.py
:000000 100644 00000000 34aa53bf A	excel_import_export/tests/__init__.py
:000000 100644 00000000 18618688 A	excel_import_export/tests/sale_order.xlsx
:000000 100644 00000000 c8481487 A	excel_import_export/tests/test_xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 800ea573 1807ea7e M	excel_import_export/wizard/export_xlsx_wizard.py
:100644 100644 febed8d0 750dc17e M	excel_import_export/wizard/import_xlsx_wizard.py
:100644 100644 475b5187 392fe6e5 M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 8e40a2d0 21574896 M	excel_import_export_demo/report_sale_order/report_sale_order.py
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3 fee958bc M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 9738a3c8 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 c7db3f92 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f 12f9ca99 M	excel_import_export/models/xlsx_import.py
:100644 100644 70c37799 f123d2a6 M	excel_import_export/models/xlsx_report.py
:100644 100644 58689ee5 ed8c9fc7 M	excel_import_export/models/xlsx_template.py
:000000 100644 00000000 34aa53bf A	excel_import_export/tests/__init__.py
:000000 100644 00000000 18618688 A	excel_import_export/tests/sale_order.xlsx
:000000 100644 00000000 69aa6ea0 A	excel_import_export/tests/test_xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 800ea573 1807ea7e M	excel_import_export/wizard/export_xlsx_wizard.py
:100644 100644 febed8d0 750dc17e M	excel_import_export/wizard/import_xlsx_wizard.py
:100644 100644 475b5187 392fe6e5 M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 8e40a2d0 21574896 M	excel_import_export_demo/report_sale_order/report_sale_order.py
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml

:100644 100644 96157ea3 fee958bc M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 9738a3c8 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 c7db3f92 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f 933d8614 M	excel_import_export/models/xlsx_import.py
:100644 100644 70c37799 f123d2a6 M	excel_import_export/models/xlsx_report.py
:100644 100644 58689ee5 1460473a M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 800ea573 1807ea7e M	excel_import_export/wizard/export_xlsx_wizard.py
:100644 100644 febed8d0 750dc17e M	excel_import_export/wizard/import_xlsx_wizard.py
:100644 100644 a2d035ef 9463f279 M	excel_import_export_demo/__manifest__.py
:100644 100644 475b5187 e7f1255b M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 8e40a2d0 21574896 M	excel_import_export_demo/report_sale_order/report_sale_order.py
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml
:000000 100644 00000000 79db62f7 A	excel_import_export_demo/tests/__init__.py
:000000 100644 00000000 18618688 A	excel_import_export_demo/tests/sale_order.xlsx
:000000 100644 00000000 c9733b95 A	excel_import_export_demo/tests/test_common.py
:000000 100644 00000000 9c943768 A	excel_import_export_demo/tests/test_xlsx_import_export.py
:000000 100644 00000000 730605c1 A	excel_import_export_demo/tests/test_xlsx_template.py

:100644 100644 96157ea3 fee958bc M	excel_import_export/__manifest__.py
:100644 100644 00ee3d9f 51c2572a M	excel_import_export/models/common.py
:100644 100644 a215d29b 9738a3c8 M	excel_import_export/models/styles.py
:100644 100644 ace11a32 c7db3f92 M	excel_import_export/models/xlsx_export.py
:100644 100644 881b814f 933d8614 M	excel_import_export/models/xlsx_import.py
:100644 100644 70c37799 f123d2a6 M	excel_import_export/models/xlsx_report.py
:100644 100644 58689ee5 1460473a M	excel_import_export/models/xlsx_template.py
:100644 100644 5c9c09a6 a363ad19 M	excel_import_export/views/xlsx_template_view.xml
:100644 100644 800ea573 1807ea7e M	excel_import_export/wizard/export_xlsx_wizard.py
:100644 100644 febed8d0 750dc17e M	excel_import_export/wizard/import_xlsx_wizard.py
:100644 100644 a2d035ef 9463f279 M	excel_import_export_demo/__manifest__.py
:100644 100644 475b5187 e7f1255b M	excel_import_export_demo/import_export_sale_order/templates.xml
:100644 100644 8e40a2d0 21574896 M	excel_import_export_demo/report_sale_order/report_sale_order.py
:100644 100644 4af9c519 45ee33c6 M	excel_import_export_demo/report_sale_order/templates.xml
:000000 100644 00000000 79db62f7 A	excel_import_export_demo/tests/__init__.py
:000000 100644 00000000 18618688 A	excel_import_export_demo/tests/sale_order.xlsx
:000000 100644 00000000 bb3ea32e A	excel_import_export_demo/tests/test_common.py
:000000 100644 00000000 9c943768 A	excel_import_export_demo/tests/test_xlsx_import_export.py
:000000 100644 00000000 730605c1 A	excel_import_export_demo/tests/test_xlsx_template.py

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update excel_import_export.pot

kittiu committed:
[12.0][FIX] excel_import_export

wrong use of dict.iteritems() for python3

OCA-git-bot committed:
[ADD] icon.png

kittiu committed:
[12.0][FIX] excel_import_export

Fix Import Excel wizard, cannot download template as in v12
core Odoo prevent download default binary.

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update excel_import_export.pot

OCA-git-bot committed:
[UPD] README.rst

kittiu committed:
[12.0][FIX] excel_import_export, use sudo() to get param

kittiu committed:
[12.0][IMP] excel_import_export, excel_import_export_demo

Add report action feature and new examples

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update excel_import_export.pot

OCA-git-bot committed:
[UPD] README.rst

kittiu committed:
[12.0][FIX] excel_import_export, fix wrong file name on report_action

When mark to convert xlsx to csv, csv file name is not used

Saran440 committed:
[12.0][FIX] excel_import_export - Style sum footer to number

oca-travis committed:
[UPD] Update excel_import_export.pot

OCA-git-bot committed:
[UPD] README.rst

liweijie0812 committed:
Added translation using Weblate (Chinese (Simplified))

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 2.9% (5 of 170 strings)

Translation: server-tools-12.0/server-tools-12.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-excel_import_export/zh_CN/

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 24.1% (41 of 170 strings)

Translation: server-tools-12.0/server-tools-12.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-excel_import_export/zh_CN/

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (170 of 170 strings)

Translation: server-tools-12.0/server-tools-12.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-excel_import_export/zh_CN/

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (170 of 170 strings)

Translation: server-tools-12.0/server-tools-12.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-excel_import_export/zh_CN/

OCA-git-bot committed:
[UPD] README.rst

kittiu committed:
[12.0][ENH] excel_import_export, faster export

Previously export is slow on large number of rows.
This fix ensure that sheet.insert_rows is called only once for all rows,
instead of insert 1 row for every rows

oca-travis committed:
[UPD] Update excel_import_export.pot

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-12.0/server-tools-12.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-excel_import_export/

oca-travis committed:
[UPD] Update excel_import_export.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-12.0/server-tools-12.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-excel_import_export/

kittiu committed:
[12.0][FIX] excel_import_export, problem on import with datetime field

Can't import if the import file has date time field

oca-travis committed:
[UPD] Update excel_import_export.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-12.0/server-tools-12.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-excel_import_export/

kittiu committed:
[IMP] : black, isort, prettier

kittiu committed:
[13.0][MIG] excel_import_export, excel_import_export_demo

kittiu committed:
[IMP] excel_import_export: black, isort, prettier

kittiu committed:
[14.0][MIG] excel_import_export, excel_import_export_demo

oca-travis committed:
[UPD] Update excel_import_export.pot

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update excel_import_export.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-14.0/server-tools-14.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-excel_import_export/

oca-travis committed:
[UPD] Update excel_import_export.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-14.0/server-tools-14.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-excel_import_export/

alessandro-fiorino committed:
Added translation using Weblate (Italian)

alessandro-fiorino committed:
Translated using Weblate (Italian)

Currently translated at 5.5% (11 of 200 strings)

Translation: server-tools-14.0/server-tools-14.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-excel_import_export/it/

alessandro-fiorino committed:
Translated using Weblate (Italian)

Currently translated at 82.5% (165 of 200 strings)

Translation: server-tools-14.0/server-tools-14.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-excel_import_export/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 82.5% (165 of 200 strings)

Translation: server-tools-14.0/server-tools-14.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-excel_import_export/it/

OCA-git-bot committed:
[IMP] update dotfiles

Commit:
[15,0][MIG] excel_import_export, excel_import_export_demo:

migration to 15.0. Changes by pre-commit errors and warnings,
changes by original Odoo code changes, JS files suggestions.

alessandro-fiorino committed:
Translated using Weblate (Italian)

Currently translated at 82.5% (165 of 200 strings)

Translation: server-tools-14.0/server-tools-14.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-excel_import_export/it/

luisg123v committed:
[REF] excel_import_export: remove xlrd & xlwt from external dependencies

Because those dependencies are already part of Odoo's base dependencies.

OCA-git-bot committed:
excel_import_export 14.0.1.1.0

ibuioli committed:
Added translation using Weblate (Spanish (Argentina))

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (200 of 200 strings)

Translation: server-tools-14.0/server-tools-14.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-excel_import_export/es_AR/

Commit:
[15.0][FIX] excel_import_export, excel_import_export_demo:

fix from PR #2283 together with @luc-demeyer suggestion https://github.com/OCA/server-tools/pull/2283#pullrequestreview-881026019

AungKoKoLin1997 committed:
[MIG] excel_import_export, excel_import_export_demo: Migration to 16.0

AungKoKoLin1997 committed:
[FIX] Not overriding the create method during installation

Commit:
[UPD] Update excel_import_export.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 81.1% (155 of 191 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 81.1% (155 of 191 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 81.1% (155 of 191 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

Commit:
[UPD] Update excel_import_export.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 81.2% (156 of 192 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (192 of 192 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/es/

Commit:
[UPD] Update excel_import_export.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/

OCA-git-bot committed:
[UPD] README.rst

mymage committed:
Translated using Weblate (Italian)

Currently translated at 81.2% (156 of 192 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 81.2% (156 of 192 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 81.2% (156 of 192 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

RodrigoBM committed:
[REF] excel_import_export: hooks are added so that they can be inherited

mymage committed:
Translated using Weblate (Italian)

Currently translated at 81.2% (156 of 192 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

OCA-git-bot committed:
[BOT] post-merge updates

PauBForgeFlow committed:
[IMP] excel_import_export: Set import wizard filename

Commit:
[UPD] Update excel_import_export.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (193 of 193 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/es/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 80.8% (156 of 193 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 86.5% (167 of 193 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (193 of 193 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (193 of 193 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

Commit:
[UPD] Update excel_import_export.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (194 of 194 strings)

Translation: server-tools-16.0/server-tools-16.0-excel_import_export
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-excel_import_export/it/

Josep-s73 committed:
[IMP] excel_import_export: pre-commit auto fixes

Josep-s73 committed:
[MIG] excel_import_export: Migration to 17.0

AungKoKoLin1997 committed:
[FIX] excel_import_export: remove sudo() when importing record

This commit fixes issue of finding multiple records across all companies during import when they have same record name.
Previously, importing with sudo could bypass multi-company record rules.

sanderlienaerts commented:
Functional check is working as expected 👍

sanderlienaerts commented:
Please note the following access rule warnings:
```
odoo.addons.base.models.ir_model: Rule xlsx_template_user has no group, this is a deprecated feature. Every access-granting rule should specify a group. 
odoo.addons.base.models.ir_model: Rule xlsx_template_export_user has no group, this is a deprecated feature. Every access-granting rule should specify a group. 
odoo.addons.base.models.ir_model: Rule xlsx_template_import_user has no group, this is a deprecated feature. Every access-granting rule should specify a group. 

```

sanderlienaerts commented on excel_import_export/wizard/import_xlsx_wizard.py:
```diff
@@ -0,0 +1,157 @@
+            "res_id": self.id,
+            "views": [(False, "form")],
+            "target": "new",
+        }
```
The action is missing a name (title), therefore after continuing by executing `action_import()`. The window title is no longer the export's name but `Odoo`.

sanderlienaerts commented on excel_import_export/security/ir.model.access.csv:
```diff
@@ -0,0 +1,7 @@
+xlsx_template_import_user,xlsx_template_import_user,model_xlsx_template_import,,1,1,1,1
+access_export_xlsx_wizard,access_export_xlsx_wizard,model_export_xlsx_wizard,base.group_user,1,1,1,1
+access_import_xlsx_wizard,access_import_xlsx_wizard,model_import_xlsx_wizard,base.group_user,1,1,1,1
+access_report_xlsx_wizard,access_report_xlsx_wizard,model_report_xlsx_wizard,base.group_user,1,1,1,1
```
The current definition of the security rules is deprecated. See the warnings below.
```
odoo.addons.base.models.ir_model: Rule xlsx_template_user has no group, this is a deprecated feature. Every access-granting rule should specify a group. 
odoo.addons.base.models.ir_model: Rule xlsx_template_export_user has no group, this is a deprecated feature. Every access-granting rule should specify a group. 
odoo.addons.base.models.ir_model: Rule xlsx_template_import_user has no group, this is a deprecated feature. Every access-granting rule should specify a group. 
```

sanderlienaerts commented on excel_import_export/models/__init__.py:
```diff
@@ -0,0 +1,23 @@
+#     2, (now() at time zone 'UTC'), '2020-10-05 09:39:28',
+#     NULL, '[FURN_0269] Office Chair Black', 8,
+#     '11111.00', '5.000', 1,
+#     10)
```
We should consider removing these comments

Josep-s73 commented:
> Functional check is working as expected 👍

Quick question: I see that many PRs are failing in the same module test (ERROR odoo odoo.addons.tracking_manager.tests.test_tracking_manager: FAIL: TestTrackingManager.test_not_tracked      and      ERROR odoo odoo.addons.tracking_manager.tests.test_tracking_manager: FAIL: TestTrackingManager.test_update_tracked ). Do you know what's happening here?

sanderlienaerts commented:
@pedrobaeza hi Pedro, could you attend Josep's comment about the failing tests? Any idea?

@Josep-s73 one last request would be to squash the administrative comments, as guided here: https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests#mergesquash-the-commits-generated-by-bots-or-weblate

pedrobaeza commented:
Rebase the PR after #3020 is merged.

/ocabot migration excel_import_export

AungKoKoLin1997 committed:
[IMP] excel_import_export: improve error showing

gaikaz commented on excel_import_export/wizard/import_xlsx_wizard.py:
```diff
@@ -0,0 +1,158 @@
+        string="Template",
+        required=True,
+        ondelete="cascade",
+        domain=lambda self: self._context.get("template_domain", []),
```
This template filtering doesn't seem to work.
When you have `sale.order` and `account.move` (invoice) templates, this should filter out specific templates according to the model, but instead shows all of them on both Sales and Invoices (see images below)

While testing, sometimes I would also get `No template found` Validation Error that comes from `default_get` method. 

There must be something wrong with context generation and passing.

![image](https://github.com/user-attachments/assets/d6a774b9-e42e-4eea-ab10-c6061dcaa095)

In debug mode, you can see that the domain isn't even set.
![image](https://github.com/user-attachments/assets/63332d5e-c73e-4a11-9688-2b812b7d47d1)

It would be very appreciated if you would add a test case for this. Mainly, checking if `template_domain` is generated and passed correctly.

gaikaz reviewed (commented):
Thank you for your work migrating this module. 🙏🏻 

Code and Functional review.
Testing was done on runboat environment of this PR.

AungKoKoLin1997 committed:
[IMP] excel_import_export: excel_import_export: allow selection of all related templates for export action

Before this commit, when two templates exist for the same model, clicking "ADD EXPORT ACTION" for each template
creates separate server actions for each. As a result, if multiple templates are created for the same model,
numerous server actions with the label "Export Excel" will accumulate for that model.
With this commit, if two templates exist for the same model and "ADD EXPORT ACTION" is clicked for each,
they will now share a single server action. Users can then select templates from within the "Export Excel" wizard.
Clicking "REMOVE EXPORT ACTION" on a template will remove only that template from the "Export Excel" selection.
The server action itself will be deleted only when no templates use it for export.

AungKoKoLin1997 committed:
[IMP] excel_import_export: update readme

nobuQuartile commented:
Thank you for your contributions to the OCA.
Please rebase to include the recent changes regarding the 16.0 excel_import_export. Note that there are multiple merged PRs related to 16.0 excel_import_export.
Lastly, thank you for advancing the migration work.

https://github.com/OCA/server-tools/pull/3132
https://github.com/OCA/server-tools/pull/3114
https://github.com/OCA/server-tools/pull/3035
https://github.com/OCA/server-tools/pull/2992

arantxa-s73 commented:
@gaikaz @nobuQuartile Added the commented commits, and also fixed the code so that when the wizard opens we only have the model templates

You can check it

gaikaz commented:
I either forgot how to use the module or something isn't right with template getting.
Even tho I can add the action to import on `sale.order` model, whenever I click on the action, it gives this error, as if the template (the one that was added to actions) still isn't compatible?
<img width="680" height="214" alt="image" src="https://github.com/user-attachments/assets/b036fe7f-9e8b-4bc8-801c-2b3b7d881515" />

arantxa-s73 commented:
As you have it configured, I have the following and when I export, only that template appears, even though I have more.
<img width="1784" height="723" alt="image" src="https://github.com/user-attachments/assets/604b642d-5d59-4073-bc84-b88192791c66" />

gaikaz commented:
Oh, I've tested the import. I suspect, I'm missing something.

Reyes4711-S73 reviewed (approved):
LGTM
Functional review ok
